.github/workflows/test-tpg.yml

```diff
@@ -96,7 +96,7 @@ jobs:
         cd provider
         make docscheck
     - name: Post Result Status to Pull Request
-      if: ${{ !cancelled() && steps.pull_request.outputs.has_changes == 'true' }}
```

Comment:
Its certainly possible! I am leaning towards the current approach since this flow needs to happen regardless. While short circuiting with a immediate success would be slightly faster this option seems simpler from my POV. ie Skip what we don't need.